pkg/k8shandler/elasticsearch.go

```diff
@@ -440,6 +440,123 @@ func DoSynchronizedFlush(clusterName, namespace string, client client.Client) (b
 	return (payload.StatusCode == 200), payload.Error
 }
 
+func MigrateKibanaIndex(clusterName, namespace string, client client.Client) (bool, error) {
+	// check that kibana needs to be migrated
+	payload := &esCurlStruct{
+		Method: http.MethodGet,
+		URI:    ".kibana/_mapping",
+	}
+
+	curlESService(clusterName, namespace, payload, client)
+
+	// check for 404, means index does not exist -- no migration needed
+	if payload.StatusCode == 404 {
+		return true, nil
+	}
+
+	if indexBody, ok := payload.ResponseBody[".kibana"].(map[string]interface{}); ok {
+		if mappings, ok := indexBody["mappings"].(map[string]interface{}); ok {
+			if len(mappings) > 1 {
+				// if .kibana_1 doesn't exist, reindex
+				payload := &esCurlStruct{
+					Method: http.MethodGet,
+					URI:    ".kibana_1",
+				}
+
+				curlESService(clusterName, namespace, payload, client)
+
+				if payload.StatusCode == 404 {
+					// reindex to .kibana_1
+					payload := &esCurlStruct{
+						Method:      http.MethodPost,
+						URI:         "_reindex",
+						RequestBody: "{\"source\":{\"index\":\".kibana\"},\"dest\":{\"index\":\".kibana_1\",\"type\":\"doc\"}}",
```

Comment:
Shouldn't the new (and single) document type be `_doc` with leading underscore?
Out of curiosity, is the `"type"` field supported for `"dest"` index in `_reindex` API in ES 5.x? I never tried this.

Reply:
I set it to `doc` since that is what the kibana6 index uses when it is created.
And yup! I took the example from the ES docs and tested the reindex manually first.